lib/Form/DisplayFormInWPContent.php

```diff
@@ -103,15 +100,28 @@ private function saveNoForms() {
   }
 
   /**
-   * @return FormEntity[]
+   * @return array<string, FormEntity>
    */
   private function getForms(): array {
     $forms = $this->formsRepository->findBy(['deletedAt' => null]);
-    return array_filter($forms, function($form) {
-      return (
-        $this->shouldDisplayForm($form)
-      );
-    });
+    $forms = $this->filterOneFormInEachDisplayType($forms);
```

Comment:
I think you need to add some order here to ensure that the latest updated will be selected.